bindings/node/src/tokenizer.rs

```diff
@@ -0,0 +1,540 @@
+use crate::decoders::Decoder;
+use crate::encoding::{JsEncoding, JsTruncationDirection, JsTruncationStrategy};
+use crate::models::Model;
+use crate::normalizers::Normalizer;
+use crate::pre_tokenizers::PreTokenizer;
+use crate::processors::Processor;
+use crate::tasks::tokenizer::{DecodeBatchTask, DecodeTask, EncodeBatchTask, EncodeTask};
+use crate::trainers::Trainer;
+use std::collections::HashMap;
+use tokenizers::Model as ModelTrait;
+
+use napi::bindgen_prelude::*;
+use napi_derive::napi;
+use std::sync::{Arc, RwLock};
+use tokenizers as tk;
+
+#[napi]
+#[derive(Default)]
+pub enum PaddingDirection {
+  #[default]
+  Left,
+  Right,
+}
+
+impl From<PaddingDirection> for tk::PaddingDirection {
+  fn from(w: PaddingDirection) -> Self {
+    match w {
+      PaddingDirection::Left => tk::PaddingDirection::Left,
+      PaddingDirection::Right => tk::PaddingDirection::Right,
+    }
+  }
+}
+
+impl TryFrom<String> for PaddingDirection {
+  type Error = Error;
+  fn try_from(w: String) -> Result<Self> {
+    match w.as_str() {
+      "left" => Ok(PaddingDirection::Left),
+      "right" => Ok(PaddingDirection::Right),
+      s => Err(Error::from_reason(format!(
+        "{s:?} is not a valid direction"
+      ))),
+    }
+  }
+}
+
+#[napi(object)]
+#[derive(Default)]
+pub struct PaddingOptions {
+  pub max_length: Option<u32>,
+  pub direction: Option<Either<String, PaddingDirection>>,
+  pub pad_to_multiple_of: Option<u32>,
+  pub pad_id: Option<u32>,
+  pub pad_type_id: Option<u32>,
+  pub pad_token: Option<String>,
+}
+
+impl TryFrom<PaddingOptions> for tk::PaddingParams {
+  type Error = Error;
+  fn try_from(value: PaddingOptions) -> Result<Self> {
+    let direction = match value.direction {
+      Some(either) => match either {
+        Either::A(string) => {
+          let direction: PaddingDirection = string.try_into()?;
+          direction.into()
+        }
+        Either::B(direction) => direction.into(),
+      },
+      None => tk::PaddingDirection::Right,
+    };
+    Ok(Self {
+      pad_to_multiple_of: value.pad_to_multiple_of.map(|s| s as usize),
+      pad_id: value.pad_id.unwrap_or_default(),
+      pad_type_id: value.pad_type_id.unwrap_or_default(),
+      pad_token: value.pad_token.unwrap_or("[PAD]".to_string()),
+      direction,
+      strategy: match value.max_length {
+        Some(length) => tk::PaddingStrategy::Fixed(length as usize),
+        None => tk::PaddingStrategy::BatchLongest,
+      },
+    })
+  }
+}
+
+#[napi(object)]
+#[derive(Default)]
+pub struct EncodeOptions {
+  pub is_pretokenized: Option<bool>,
+  pub add_special_tokens: Option<bool>,
+}
+
+#[derive(Default)]
+struct EncodeOptionsDef {
+  // TODO
```

Comment:
TODO in a subsequent PR?

Reply:
*Maybe* see comment above on pretokenized sequences I discarded for typing reasons. (Re-adding will be non breaking)